[Docs] Update docs about App Bridge usage (#945)
Shopify/polaris-react: src/components/AppProvider/README.md

## Initializing the Shopify App Bridge

> When using Polaris, you don’t need to go through the initialization of the Shopify App Bridge as described [in the docs](https://help.shopify.com/en/api/embedded-apps/app-bridge#set-up-your-app). Instead, configure the connection to the Shopify admin through the app provider component, which must wrap all components in an embedded app. This component initializes the Shopify App Bridge using the `apiKey` and `shopOrigin` you provide. **The `apiKey` attribute is required** and can be found in the [Shopify Partner Dashboard](https://partners.shopify.com).

Reviewer: ```suggestion
When using Polaris, you don’t need to go through the initialization of the Shopify App Bridge as described in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/app-bridge#set-up-your-app). Instead, configure the connection to the Shopify admin through the [app provider component](https://polaris.shopify.com/components/structure/app-provider), which wraps all components in an embedded app. The app provider component initializes the Shopify App Bridge using the `apiKey` and `shopOrigin` that you provide. 

**The `apiKey` attribute is required**. You can find the API key for your app in the **Apps** section of your [Shopify Partner Dashboard](https://partners.shopify.com).
```

- A few style adjustments and an added cross-link

> **The `shopOrigin` attribute is optional**, however it is best practice to [get and store `shopOrigin` yourself](https://help.shopify.com/en/api/embedded-apps/shop-origin).

Reviewer: - Question about this! Is it a best practice to include the `shopOrigin` attribute? Or just that, if you DO include it, that you get and store it manually?

Reviewer: ```suggestion
**The `shopOrigin` attribute is optional, but recommended**.  Learn how to get and store the shop origin in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/shop-origin).
```

Reviewer 2: i think we should not even mention that it's optional, because it really isn't (despite the props contract)

> **The `shopOrigin` attribute is optional**, however it is best practice to get and store `shopOrigin` by yourself which can be found in [Embedded apps docs](https://help.shopify.com/en/api/embedded-apps/shop-origin).

Reviewer 3: Maybe we can simplify? 

> The `shopOrigin` attribute is optional**,  however it is best practice to [get and store `shopOrigin` yourself](https://help.shopify.com/en/api/embedded-apps/shop-origin).

## Initializing the Shopify App Bridge

> When using Polaris, you don’t need to go through the initialization of the Shopify App Bridge as described in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/app-bridge#set-up-your-app). Instead, configure the connection to the Shopify admin through the [app provider component](https://polaris.shopify.com/components/structure/app-provider), which wraps all components in an embedded app. The app provider component initializes the Shopify App Bridge using the `apiKey` and `shopOrigin` that you provide.

Reviewer 4: ```suggestion
When using Polaris, you don’t need to go through the initialization of the Shopify App Bridge as described in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/app-bridge#set-up-your-app). Instead, configure the connection to the Shopify admin through the [app provider component](https://polaris.shopify.com/components/structure/app-provider), which wraps all components in an embedded app. The app provider component initializes the Shopify App Bridge using the `apiKey` and `shopOrigin` that you provide. **The `apiKey` and the `shopOrigin` attributes are required.** Find the API key for your app in the Apps section of your [Shopify Partner Dashboard](https://partners.shopify.com). Learn how to get and store the shop origin in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/shop-origin).
```

> **The `apiKey` attribute is required**. You can find the API key for your app in the **Apps** section of your [Shopify Partner Dashboard](https://partners.shopify.com).

Reviewer 4: ```suggestion
**The `apiKey` attribute is required**. You can find the API key for your app in the Apps section of your [Shopify Partner Dashboard](https://partners.shopify.com).
```

Reviewer 4: Small tweak to unbold the page section name so it matches the formatting of the rest of the style guide

> **The `shopOrigin` attribute is required**. Learn how to get and store the shop origin in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/shop-origin).

Author: Based on @gf3's comment, we should state `shopOrigin` as required field despite the props contract.

Author: Do you have any thoughts before shipping this PR? @sadiesaurus @selenehinkley

Reviewer 4: Hey @henrytao-me! Sadie and I edited that last bit to condense it, while still getting the message accross that the `apiKey` and `shopOrigin` are required. If you like what we updated it to, it's all good to 🚢

> **The `apiKey` attribute is required**. You can find the API key for your app in the Apps section of your [Shopify Partner Dashboard](https://partners.shopify.com).

Reviewer 4: ```suggestion
```

> **The `shopOrigin` attribute is required**. Learn how to get and store the shop origin in the [Shopify Help Center](https://help.shopify.com/en/api/embedded-apps/shop-origin).

Reviewer 4: ```suggestion
```
